rasterio/windows.py

```diff
@@ -303,10 +302,71 @@ def shape(window, height=-1, width=-1):
     values in the window.
     """
     (a, b), (c, d) = evaluate(window, height, width)
-    return b-a, d-c
+    return (b - a, d - c)
 
 
 def window_index(window):
     # "window_" is necessary here to redundancy to disambiguate
     # from transform.get_index and src.index
     return tuple(slice(*w) for w in window)
+
+
+class Window(tuple):
+    """Windows are rectangular subsets of rasters"""
```

Comment:
Should we explain in the docstring how the constructor (offset/num style) differs from the internal representation (startp/stop ranges)?